packages/spector/src/cli/cli.ts

```diff
@@ -14,9 +15,56 @@ import { getCommit } from "../utils/misc-utils.js";
 
 export const DEFAULT_PORT = 3000;
 
+function copyDir(src: string, dest: string) {
```

Comment:
can we not just run it from where they are defined. This bring a lot of variability and makes it very hard to test any fix if the error show up in a temp directory

Reply:
I have removed the `copyDir` and optimized the code.

Reply:
Sorry I meant why do you copy things. We should run from the dir they are in

Reply:
Copying in a different dir is likely to break imports as well

Reply:
The reason why I copied the files is there are scripts such as `validate-scenarios`, etc will error out if they encounter any error in one of the scenario paths. Even the coverage was based on single commit. 

But, You are right. There is a possibility that the imports might break. With the recent update of PR, I have executed all scenarios as is in the current location.

Reply:
@timotheeguerin One important thing to note is the structure of `spec-coverage.json` has been modified to an array. So, this requires down the line changes. (Dashboard???)

Reply:
Can we not merge it as right now they are all unique. Otherwise I think the new structure should be a 2 level record with the first key being the test suite

Reply:
@timotheeguerin Even with the new structure they are still unique. It looks like this:

![image](https://github.com/user-attachments/assets/9fe6c3c3-e304-410a-b435-75b305f67b3e)

So, Do you want to do any changes in this?

Reply:
Eh I guess that's ok for now, we'll update when we update the dashboard